test/common/index.js

```diff
@@ -690,12 +690,26 @@ Object.defineProperty(exports, 'hasIntl', {
 });
 
 // Useful for testing expected internal/error objects
+const kAcceptableErrorTypes = ['NodeError', 'AssertionError', 'RangeError'];
 exports.expectsError = function expectsError({code, type, message}) {
-  return function(error) {
+  const expectedTypeName = type && type.name;
+  return function expectsErrorCheck(error) {
     assert.strictEqual(error.code, code);
-    if (type !== undefined)
-      assert(error instanceof type,
-             `${error} is not the expected type ${type}`);
+    if (expectedTypeName) {
```

Comment:
I'm with @cjihrig.
```js
assert(err instanceof Error)
```
Does not protect for a change in the type of `err`. The code could change it from `Error` to `TypeError` and the test would pass. Even changes from `TypeError` to `RangeError` would pass.
Those are `semver-major` changes that we don't assert.
```js
coreAPI.function(arg, (err, ret) => {
  if (err instanceof TypeError) console.log('bad programer!');
  if (err instanceof Error) console.log('oops');
}
```